src/useApproveApplicationMutation.js

```diff
@@ -0,0 +1,25 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { abi, address } from '@vderunov/whitelist-contract/deployments/11155420/Whitelist';
+import { Contract } from 'ethers';
+import { useSynthetix } from './useSynthetix';
+
+function useApproveApplicationMutation() {
+  const [synthetix] = useSynthetix();
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: async (wallet) => {
+      const contract = new Contract(address, abi, synthetix.signer);
+      const tx = await contract.approveApplication(wallet);
+      await tx.wait();
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: [synthetix.chainId, synthetix.walletAddress, 'permissions'],
```

Comment:
Need to invalidate permissions for the wallet, not the admin that approves application

Cannot do `onSuccess`, but rather do it directly in the async function.

```suggestion
        queryKey: [synthetix.chainId, wallet, 'permissions'],
```

Reply:
Yes, my mistake (.
Moved invalidateQueries to an async function and removed `onSuccess` as redundant.